### ISSUE TITLE
NSFW imgprime.com

### ISSUE THREAD

**Issue by notmoter8** [uBlockOrigin/uAssets#704]
### URL(s) where the issue occurs

NSFW: `http://imgprime.com/img-59c0141d811f0.html`
edit: NSFW `http://imgprime.com/img3-59c0141d811f0.html` required to work too

### Describe the issue

Many animated ads are being shown around one button which is needed to get to the next page. Tried to manually filter them out, but the divs containing the images remain, their click handlers continue to exist too.

After applying the filter and going to the next page to see the thumbnails, it gets blocked out because of a css selector, precisely `:root center:nth-of-type(3)` which has a `display: none !important;` on it... No idea what ublock can do here.

### Screenshot(s)

NSFW - Pixelated: `https://i.imgur.com/72cQRdn.png`

### Versions

- Browser/version: [here]
- uBlock Origin version: 1.14.8

### Settings

- No change

### Notes

Blocking everything from `http://imgprime.com/bcler.php` seems to solve the issue. Images get delivered through that. But the elements containing these images are still clickable, so if you are not careful you still click on the ads.

Seems like it's using https://github.com/sanosay/exads-adblock / ExoLoader by the looks of what bcler.php returns but whatever.

**Comment by SMed79** [comment]
add
`||imgprime.com^$script`

**Comment by notmoter8** [comment]
`||imgprime.com^$script` by itself is unfortunately not a fix, because the image is not being shown (on the next url, NSFW: `http://imgprime.com/img3-59c0141d811f0.html`)

**Comment by okiehsch** [comment]
Does
`imgprime.com##script:inject(abort-on-property-write.js, document.createElement)`
work for you?

**Comment by notmoter8** [comment]
Getting this error on the second site with that line.

```
Uncaught TypeError: document.createElement is not a function
    at Object.detectCensorship (frler.js:2)
    at Object.addZone (frler.js:20)
    at img3-59c0141d811f0.html:64
```

**Comment by okiehsch** [comment]
Well, they are expected because `http://imgprime.com/frler.js` is the Exolader script that pulls the ads in the first place. They do not result in any site breakage as far as I can see.
`imgprime.com##script:inject(abort-on-property-read.js, document.createElement)` will also work and 
should not result in any errors, so if you prefer that I will add that one.

**Comment by notmoter8** [comment]
@okiehsch unfortunately the image at NSFW: `http://imgprime.com/img3-59c0141d811f0.html` does not show up (it's hidden by a few css rules and a `hidden` html property), so the filter as of now would just break the site. Any idea how to get the image to show up?

**Comment by okiehsch** [comment]
Hm, the image at `http://imgprime.com/img3-59c0141d811f0.html` shows on my end, so I can not help.
Does it  never show up on your end or only if you add
`imgprime.com##script:inject(abort-on-property-read.js, document.createElement)`, or
`imgprime.com##script:inject(abort-on-property-write.js, document.createElement)`?

**Comment by okiehsch** [comment]
I tried with Chrome and Firefox, the image always shows up on my end.

**Comment by notmoter8** [comment]
Those 2 filters don't seem to help. On firefox the image gets shown, but on Chrome `Version 60.0.3112.113`it doesn't :(

**Comment by okiehsch** [comment]
I use Chrome stable version `61.0.3163.91` and  it works here.

**Comment by mapx-** [comment]
Did you try disabling all other extensions ?

**Comment by gotitbro** [comment]
@notmoter8 Disable your My Filters list.

**Comment by gotitbro** [comment]
They do not result in any site breakage as far as I can see.
What do you mean by that @okiehsch ?

The ads get blocked by `||imgprime.com/frler.js` also with the `abort-on-property` filters.

A sister site to this website: `http://imgking.co/imgv-5922e3366cf7f.html` (NSFW)

**Comment by okiehsch** [comment]
@gotitbro I mean the `Uncaught TypeError: document.createElement is not a function`, that you will see if you add `imgprime.com##script:inject(abort-on-property-write.js, document.createElement)` and open the Chrome console will not break any site functionality, specifically the image loads fine on my end.

So does
````
imgking.co,imgsking.com,imgprime.com,newpornup.com##script:inject(abort-on-property-read.js, document.createElement)
````
work for you?

**Comment by notmoter8** [comment]
@okiehsch With that filter the original links seem to work just fine now (I also deleted all other filters I had, that may have solved it).

But now, the image at **NSFW** `http://imgsking.com/image/7X1` seems to be blocked. Removing that domain from the filter blocks its ads just fine (it was working properly without it).


Tldr: add `imgking.co,imgprime.com,newpornup.com##script:inject(abort-on-property-read.js, document.createElement)` to the filter list and this issue is closed 👍

**Comment by gotitbro** [comment]
> I also deleted all other filters I had, that may have solved it

Yup, you had mentioned in this thread many conflicting filters removing them let the suggested filters work.
